### ISSUE TITLE
[Size Only]:Add configuration option to make New-* cmdlets non-idempotent

### ISSUE THREAD

**Issue by markcowl** [Azure/autorest.powershell#569]
Some APIs, like AzureStack Admin APIs use PUT for create, but do not support headers like 
```
if-none-match: *
```

That would force the PUT to fail if the resource already existed.

- Determine a high-level design for how we could make 'New-*' cmdlets that used such a PUT operation non-idempotent (that is, make them fail if the resource already exists)
- Determine a rough cost for implementing this in the generator

- The *brute force* method would be to compose `GET` and `PUT` operations for the resource

``` powershell
$foo = Get-AzWidget <params>
If ($foo -ne $null)
{
    throw new PSInvalidOperationException("Widget<name> already exists")
}

New-AzWidget <params>
```

**Comment by dolauli** [comment]
@markcowl I really do not like this idea. New-* is mapped to http PUT, which is idempotent by default. If RP expects failing on existing, they should implement "if-none-match" as you mentioned. BTW, check existence before the creation are two steps. We can not guarantee that nobody create the resource after the check and before the creation. If some one really want to do this, I suggest they should do it through customization. And this case is not a common one I think.

**Comment by markcowl** [comment]
@dolauli  I agree on both points.  The issue is that this is a breaking change per their existing cmdlets, so the ask is to just size the effort so we can get an estimate of this versus all the documentation updated or customizations
